<launch>
  <arg name="voc_file" default="$(find ORB_SLAM2)/Vocabulary/ORBvoc.bin"/>
  <arg name="settings_file" default="$(find ORB_SLAM2)/config/mono.yaml"/>
  <arg name="image_topic" value="/camera/rgb/image_raw"/>

  <node name="Monopub" pkg="ORB_SLAM2"
      type="Monopub" args = "$(arg voc_file) $(arg settings_file)" output="screen">

       <remap from="/camera/image_raw" to="$(arg image_topic)" />

       <param name="publish_pointcloud" type="bool" value="true" />
       <param name="publish_pose" type="bool" value="true" />
       <param name="localize_only" type="bool" value="false" />
       <param name="reset_map" type="bool" value="false" />

       <!-- static parameters -->
       <param name="load_map" type="bool" value="false" />
       <param name="map_file" type="string" value="map.bin" />
       <param name="settings_file" type="string" value="$(find ORB_SLAM2)/config/mono.yaml" />
       <param name="voc_file" type="string" value="$(find ORB_SLAM2)/Vocabulary/ORBvoc.txt" />

       <param name="pointcloud_frame_id" type="string" value="map" />
       <param name="camera_frame_id" type="string" value="camera_link" />
       <param name="min_num_kf_in_map" type="int" value="5" />
  </node>
</launch>
